src/Reader/Reader.php

```diff
@@ -207,26 +212,27 @@ public static function import($uri, $etag = null, $lastModified = null)
                     $lastModified = $cache->getItem($cacheId . '_lastmodified');
                 }
                 if ($etag) {
-                    $headers->addHeaderLine('If-None-Match', $etag);
+                    $headers['If-None-Match'] = [$etag];
                 }
                 if ($lastModified) {
-                    $headers->addHeaderLine('If-Modified-Since', $lastModified);
+                    $headers['If-Modified-Since'] = [$lastModified];
                 }
             }
-            $response = $client->send();
+            $response = $client->get($uri, $headers);
```

Comment:
It's not. The second argument is defined as *optional*, and it is up to an implementation to decide if (a) it can accept it, and (b) if it will do anything with it. This was done so as to prevent a BC break, and is in a similar vein to how we pass the `$context` argument in zend-inputfilter and zend-validator (validators are not *required* to accept it).